src/txn/txn_rollback_to_stable.c

```diff
@@ -940,42 +1016,50 @@ __rollback_abort_row_leaf(WT_SESSION_IMPL *session, WT_REF *ref, wt_timestamp_t
 
     page = ref->page;
 
+    WT_RET(__wt_scr_alloc(session, 0, &key));
+
     /*
      * Review the insert list for keys before the first entry on the disk page.
      */
     if ((insert = WT_ROW_INSERT_SMALLEST(page)) != NULL)
-        __rollback_abort_insert_list(session, insert, rollback_timestamp);
+        WT_ERR(__rollback_abort_insert_list(session, page, insert, rollback_timestamp, NULL));
 
     /*
      * Review updates that belong to keys that are on the disk image, as well as for keys inserted
      * since the page was read from disk.
      */
     WT_ROW_FOREACH (page, rip, i) {
         stable_update_found = false;
-        if ((upd = WT_ROW_UPDATE(page, rip)) != NULL)
-            __rollback_abort_update(session, upd, rollback_timestamp, &stable_update_found);
+        if ((upd = WT_ROW_UPDATE(page, rip)) != NULL) {
+            WT_ERR(__wt_row_leaf_key(session, page, rip, key, false));
+            WT_ERR(
+              __rollback_abort_update(session, key, upd, rollback_timestamp, &stable_update_found));
+        }
 
         if ((insert = WT_ROW_INSERT(page, rip)) != NULL)
-            __rollback_abort_insert_list(session, insert, rollback_timestamp);
+            WT_ERR(__rollback_abort_insert_list(session, page, insert, rollback_timestamp, NULL));
 
         /*
          * If there is no stable update found in the update list, abort any on-disk value.
          */
         if (!stable_update_found)
-            WT_RET(__rollback_abort_ondisk_kv(session, ref, NULL, rip, rollback_timestamp, 0));
+            WT_ERR(__rollback_abort_ondisk_kv(session, ref, NULL, rip, rollback_timestamp, 0));
     }
 
     /*
      * If the configuration is not in-memory, abort history store updates from the reconciled pages
      * of data store.
      */
     if (!F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
-        WT_RET(__rollback_abort_row_reconciled_page(session, page, rollback_timestamp));
```

Comment:
There is no need for this function now. This is to remove the additional contents that went into the history store.